addons/sale_expense/models/hr_expense_split.py

```diff
@@ -31,4 +32,9 @@ def _compute_can_be_reinvoiced(self):
     @api.depends('can_be_reinvoiced')
     def _compute_sale_order_id(self):
         for split in self:
-            split.sale_order_id = split.sale_order_id if split.can_be_reinvoiced else False
+            if split.can_be_reinvoiced:
+                split.sale_order_id = split.expense_id.sale_order_id
+                split.sale_order_line_id = split.expense_id.sale_order_line_id
+                continue
```

Comment:
a bit weird (although equivalent), why not `else:` ?

Reply:
Just a reflex of optimizing indents , doesn't make sense here